libshaderc/include/shaderc/shaderc.hpp

```diff
@@ -97,6 +111,14 @@ class CompilationResult {
   shaderc_compilation_result_t compilation_result_;
 };
 
+// A compilation result for a SPIR-V binary module, which is an array
+// of uint32_t words.
+using SpvCompilationResult = CompilationResult<uint32_t>;
+// A compilation result in SPIR-V assembly syntax.
+using AssemblyCompilationResult = CompilationResult<char>;
+// Preprocessed source text.
+using PreprocessedSourceCompilationResult = CompilationResult<char>;
+
 // Contains any options that can have default values for a compilation.
 class CompileOptions {
```

Comment:
Wouldn't it make sense to also remove `SetPreprocessingOnlyMode()` and `SetDisassemblyMode()` from `CompileOptions`?